### PR TITLE
Fix#1707

### PR DESCRIPTION
**Pull request by kmonson**
# Description

Support for health subsystem in base historian. Fix minor bugs in base historian.

Fixes # 1707

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Added Unit tests.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1717%23issuecomment-399994772%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1717%23issuecomment-400008709%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1717%23discussion_r197855782%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1717%23discussion_r197856358%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1717%23issuecomment-399994772%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hold%20off%20on%20merging%20this.%20I%27ve%20got%20to%20fix%20the%20test%20imports.%22%2C%20%22created_at%22%3A%20%222018-06-25T15%3A33%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kmonson%22%7D%7D%2C%20%7B%22body%22%3A%20%22OK%2C%20it%27s%20ready%20to%20go.%22%2C%20%22created_at%22%3A%20%222018-06-25T16%3A13%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/2049915%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kmonson%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2067cbab97687d053cea56ecac128da0b82bd0cd25%20volttrontesting/services/historian/test_base_historian.py%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1717%23discussion_r197855782%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20are%20you%20using%20the%20dev%20mark%20here%3F%20%20Can%20we%20use%20a%20new%20health%20mark%3F%22%2C%20%22created_at%22%3A%20%222018-06-25T16%3A16%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%2C%20%7B%22body%22%3A%20%22or%20historian%3F%22%2C%20%22created_at%22%3A%20%222018-06-25T16%3A18%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttrontesting/services/historian/test_base_historian.py%3AL165-172%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1717#issuecomment-399994772'>General Comment</a></b>
- <a href='https://github.com/kmonson'></a> Hold off on merging this. I've got to fix the test imports.
- <a href='https://github.com/kmonson'></a> OK, it's ready to go.
- [ ] <a href='#crh-comment-Pull 67cbab97687d053cea56ecac128da0b82bd0cd25 volttrontesting/services/historian/test_base_historian.py 14'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1717#discussion_r197855782'>File: volttrontesting/services/historian/test_base_historian.py:L165-172</a></b>
- <a href='https://github.com/craig8'></a> Why are you using the dev mark here?  Can we use a new health mark?
- <a href='https://github.com/craig8'></a> or historian?

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1717?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1717?approve=1'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1717'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1717'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by kmonson** [comment]
Hold off on merging this. I've got to fix the test imports.

**Comment by kmonson** [comment]
OK, it's ready to go.

File: volttrontesting/services/historian/test_base_historian.py [hunk 1]
```diff
@@ -167,7 +165,7 @@ def historian(request, volttron_instance):
 def message_handler(peer, sender, bus,  topic, headers, message):
     alert_publishes.append(Status.from_json(message))
 
-@pytest.mark.historian
+@pytest.mark.dev
```
Comment (craig8) [inline volttrontesting/services/historian/test_base_historian.py]: Why are you using the dev mark here?  Can we use a new health mark?

Comment (craig8) [reply volttrontesting/services/historian/test_base_historian.py]: or historian?

**Pull request merged**